src/Sulu/Bundle/AdminBundle/Resources/js/containers/MultiListOverlay/MultiListOverlay.js

```diff
@@ -67,13 +68,27 @@ class MultiListOverlay extends React.Component<Props> {
             USER_SETTINGS_KEY,
             observableOptions,
             options,
+            undefined,
             preloadSelectedItems ? preSelectedItems.map((preSelectedItem) => preSelectedItem.id) : undefined
         );
+
+        this.changeOptionsDisposer = reaction(
+            () => this.props.options,
+            (options) => {
+                // reset liststore to reload whole tree instead of children of current active item
+                this.listStore.reset();
+                // set selected items as initialSelectionIds to expand them in case of a tree
+                this.listStore.initialSelectionIds = this.listStore.selectionIds;
+                this.listStore.options = {...this.listStore.options, ...options};
+            },
+            {equals: comparer.structural}
```

Comment:
Why is that needed here, but not in the `Selection` field type?

Reply:
okay so this will get kinda long 🙈 

in case of the `Selection` field type we are reacting to everything that is accessed in the `buildRequestOptions` method. The `buildRequestOptions` method accesses the observable properties of the resource store through the `formInspector.getValueByPath` method. this reaction will only run, if one of these observable properties is changed.

in case of the `MultiSelection` component (i am using that one for this example because it is higher up in the component tree, the explanation is the same for the `MultiListOverlay`), we are reacting to the value of `props.options`.  This `props.options` object is constructed in the render method of the `Selection` field type. Therefore, if the `Selection` field type is rerendered (eg. because it gets disabled), a new object will be constructed for `props.options`. 

Unfortunately, javascript evaluates `{key: 'value'} == {key: 'value'}` to `false` and therefore the new assignment of the `props.options` property will fire the reaction inside the `MultiSelection`, even if the scalar values inside the `props.options` did not change.

Reply:
I think that makes sense :slightly_smiling_face: